traits/traits.py

```diff
@@ -112,13 +112,15 @@ def multi_line_text_editor ( auto_set=True, enter_set=False ):
     """
     global MultilineTextEditor
 
-    if MultilineTextEditor is None:
+    if (auto_set, enter_set) not in MultilineTextEditor:
         from traitsui.api import TextEditor
-        MultilineTextEditor = TextEditor( multi_line = True,
-                                          auto_set   = auto_set,
-                                          enter_set  = enter_set )
+        MultilineTextEditor[(auto_set, enter_set)] = TextEditor(
```

Comment:
Nit: the parentheses are redundant here: `MultilineTextEditor[auto_set, enter_set]` would be enough. (But this is fine, too; it's really a question of taste.)